### PR TITLE
Upgrade-node-to-18

### PR DESCRIPTION
**Pull request by gyfchong**
## Why
Certain dependencies are requiring us to move to node 18

## What
- Update our `package.json` to node 18
- Update Docker to node 18

### CODE REVIEW

File: packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import React, { useEffect, useState } from "react"
 import { render, screen, waitFor } from "@testing-library/react"
 import userEvent from "@testing-library/user-event"
 import { DateRange } from "react-day-picker"
-import { DateRangePicker, DateRangePickerProps, formatDateRangeValue } from "./"
+import { formatDateRangeValue } from "./utils/formatDateRangeValue"
```
Comment (HeartSquared) [inline packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx:5]: How did this even happen 😅

File: packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx [hunk 2]
```diff
@@ -2,7 +2,8 @@ import React, { useEffect, useState } from "react"
...
+import { DateRangePicker, DateRangePickerProps } from "./"
```
Comment (HeartSquared) [inline packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx:6]:
```suggestion
import { DateRangePicker, DateRangePickerProps } from "./index"
```

Would this be safer? 🤷🏻

**Review by HeartSquared — DISMISSED** [review]

File: packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx (hunk 1 above)
Comment (gyfchong) [reply packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx:5]: Tests failed if I didn't do it this way

Comment (HeartSquared) [reply packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx:5]: Yeah it sounds like the util used to be exported from the index, but no longer does. It's just weird that it wasn't caught earlier.

Comment (gyfchong) [reply packages/date-picker/src/DateRangePicker/DateRangePicker.spec.tsx:5]: It is exported from index, but oddly doesn't pick up:
```export * from "./DateRangePicker"
export * from "./utils"
```

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**